fix: mobile navbar fix

SteamingPilot opened the pull request:
# Pull Request
Fix: Mobile NavBar
## Description
The navigation bar wasn't coded properly to handle tablet and mobile design. This PR will fix that.
I copied the **MobileNav** component from **Material Tailwind** to my own component, so that we don't actually have to npm install Material Tailwind and mess up our own stylings and CSS.

For the animation, I installed a package **framer-motion**


## Type of Change

<!-- Mark the relevant option with an "x" -->

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Code refactoring
- [ ] Performance improvement
- [ ] Accessibility improvement
- [ ] Other (please describe):

## Related Issues
None

## Code Quality Checklist

<!-- Mark completed items with an "x" -->

- [X] Code follows the project's coding standards
- [X] Self-review of the code has been performed
- [X] Code is properly commented where necessary
- [X] No console.log statements left in the code
- [X] No hardcoded values that should be configurable
- [X] Error handling is implemented where appropriate

## Testing

<!-- Mark completed items with an "x" -->

- [X] Changes have been tested locally
- [ ] All existing tests pass (if applicable)
- [ ] New tests have been added for new functionality (if applicable)
- [X] Manual testing has been performed
- [ ] Accessibility testing has been performed (for UI changes)

## Accessibility (for UI changes)

<!-- Mark completed items with an "x" -->

- [X] All interactive elements are keyboard accessible
- [X] Color contrast meets WCAG AA standards
- [ ] Images have appropriate alt text
- [ ] Form labels are properly associated
- [X] Focus indicators are visible
- [ ] Screen reader compatibility has been tested

## Build & Deployment

<!-- Mark completed items with an "x" -->

- [X] Application builds successfully
- [X] No build warnings or errors
- [X] Dependencies are up to date
- [X] No security vulnerabilities introduced

## Screenshots (if applicable)
<img width="947" height="627" alt="image" src="https://github.com/user-attachments/assets/0fb16d48-95e9-4e4f-abad-7fd75711673a" />


## Additional Notes

---

**Note**: This PR will be automatically checked for:

- Code quality (ESLint)
- Code formatting (Prettier)
- Build verification
- Accessibility compliance
- Security vulnerabilities

All checks must pass before this PR can be merged.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Implements an accessible, responsive mobile/desktop navbar with improved scrolling and adds placeholder 2025 speakers powering sessions and speakers sections.
> 
> - **Frontend**:
>   - **Navbar**: Major rewrite with accessible mobile menu (toggle button with ARIA, close-on-outside/touch handling), improved active-section detection using navbar height, smooth scrolling, separate desktop/mobile nav lists, home navigation behavior, and minor menu items tweak.
>   - **App**: Adds a skip-to-content link for accessibility.
>   - **New Component**: `components/MobileNav.jsx` (animated mobile nav using `framer-motion`).
>   - **Pages/Sections**:
>     - `Home.jsx`: Wires 2025 speakers data and track tabs into `SessionsSection` and `SpeakersSection`.
>     - `SessionsSection.jsx` / `SpeakersSection.jsx`: Add descriptive `aria-label`s to expand/collapse controls.
> - **Data**:
>   - Adds `data/2025/speakers.js` with placeholder speaker entries (with sessions) used to populate sessions and speakers.
> - **Dependencies**:
>   - Adds `framer-motion` and `@floating-ui/react` (with related subpackages).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit c5efb1c2c62865acd6076917784f8bbff5d15e47. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

SteamingPilot committed:
fix: mobile navbar fix

SteamingPilot committed:
Merge branch '2025' into mobile-navbar-fix

SteamingPilot commented:
I just noticed that there was another comment on the issue to add a 'skip-link', please wait on the review until I do that.

SteamingPilot committed:
fix: added skip-link, and aria-labels to some buttons to fix accessibility issues

SteamingPilot commented:
All set. Should be ready to merge. 

Related Issue:
Mobile NavBar treatment #29

Changes:
**Added Skip LInk**
<img width="977" height="458" alt="image" src="https://github.com/user-attachments/assets/90a3b7a9-ee44-426e-9bd0-67afee598d41" />

Works on both desktop and mobile

**Left Aligned Navigation Menu Links on mobile**
<img width="994" height="583" alt="image" src="https://github.com/user-attachments/assets/f2936b9a-8f81-41c5-ae27-7f7f7abada64" />

**Added Aria-Labels**
- Added aria-label to NavBar Hamburger Button.
- Added aria-label to Speaker Section Toggle Button. 
- Added aria-label to Session Section Toggle Button.

shrinkray commented:
# PR Review: Navigation and Mobile Issues

Thank you for your contribution! You've done a great job handling my additional items affecting the nav UX. While I appreciate the effort, this PR requires several critical fixes before it can be merged. The current implementation has significant user experience issues that need to be addressed. A lot of these problems were pre-existing, but nevertheless need to be addressed. 

## Issues Found:

1. Mobile Hover State Problem
- Issue: The `hover:z-20` class in DevTeamCard.jsx (line 15) creates problematic layering on mobile devices
- Impact: Cards may appear above navigation elements, blocking user interaction
- Fix Required: Remove `hover:z-20` from the DevTeamCard component

2. Navigation Active State Logic Flaws
The navigation system has multiple critical issues affecting user experience:
- Incorrect Active State Highlighting:
  - Clicking "Location" highlights "Sponsors" instead
  - Clicking "Sessions" highlights "Sponsors" instead
  - Clicking "Organizers" highlights "Facilitators" instead

- Broken Navigation Links:
  - "Job Board" link is non-functional (no corresponding section exists)
  - "Speakers" navigates to "Dev Team" section instead
  - These issues create a confusing and frustrating user experience

3. Missing Content Section
- Issue: "Job Board" appears in navigation but has no corresponding page section
- Impact: Users click expecting content but nothing happens
- Required Action: Either implement the Job Board section or remove it from navigation

4. Accessibility Concerns
- Issue: Active state highlighting fails contrast requirements
- Impact: Users with visual impairments cannot distinguish active navigation items
- Fix Required: Ensure active states meet WCAG contrast ratio standards
- Recommend: Using existing color and use border-bottom Tailwind class for active state

## Recommendations:

1. Fix Navigation Logic: Debug the scroll-based active state detection in Navbar.jsx
2. Remove Non-functional Links: Either implement missing sections or remove from navigation
3. Improve Mobile Experience: Remove problematic z-index hover effects
4. Enhance Accessibility: Ensure proper contrast ratios for all interactive states

## Next Steps:

Please address these issues and resubmit the PR. The navigation functionality is core to the user experience, so these fixes are essential before merging.

ShugKnight24 commented on src/layouts/SessionsSection.jsx:
```diff
@@ -82,6 +82,9 @@ const SessionsSection = ({
         />
         <button
+          aria-label={
+            isExpanded ? 'Collapse 2025 Sessions' : 'Expand 2025 Sessions'
```
Thanks for adding this in - I missed this. Can you please adjust this to dynamically account for the year with the `year` variable in a template string as this will change

ShugKnight24 replied:
Addressed in [d23ee76](https://github.com/GDG-Detroit/devfest-website/pull/40/commits/d23ee76eae8c1c5ea27d0d3694f4b0dc6ee8cbe5)

ShugKnight24 commented on src/layouts/SpeakersSection.jsx:
```diff
@@ -48,6 +48,9 @@ const SpeakersSection = ({
             className="h-8 md:h-10 lg:h-12"
           />
           <button
+            aria-label={
+              isExpanded ? 'Collapse 2025 Speakers' : 'Expand 2025 Speakers'
+            }
```
Same as above

ShugKnight24 replied:
Resolved in [d23ee76](https://github.com/GDG-Detroit/devfest-website/pull/40/commits/d23ee76eae8c1c5ea27d0d3694f4b0dc6ee8cbe5)

ShugKnight24 reviewed (commented):
Thanks for tackling this @SteamingPilot - Looks like @shrinkray gave good feedback 

Another ask on updating those aria-labels in the session and speaker sections to be dynamic

Apologies if I broke the active state on sessions and speakers. I suspect it has to do with the height of the container and I imagine it should be corrected when the real content is there

General comment, @JustinBeaudry made some changes to the nav as well, so we'll likely have to make adjustments here or in #19 to incorporate the others changes depending on which gets merged first

SteamingPilot committed:
fix: handleScroll and handleNavigation function fixed

SteamingPilot committed:
fix: made aria label on expand button dynamic for speaker and session

SteamingPilot commented:
I made some changes to fix the handleNavigation, handleScroll, and active state link, but I need to review them with @JustinBeaudry and the others before asking to merge. Will do that at the Monday meeting.

shrinkray committed:
refactor: navbar targeting

- fix hardcoded height issue
- fix active link mismatch
- improve mobile navigation
- fix multiple click issues
- separated desktop/mobile logic
- enhance ui on mobile nav
- improve scroll detection algo
- code cleanup
 - remove comments
 - add helper functions
 - fix tailwind class order
 - add some prop validation

ShugKnight24 commented on package.json:
```diff
@@ -27,7 +27,9 @@
     ]
   },
   "dependencies": {
+    "@floating-ui/react": "^0.27.16",
```
Added dependency isn't used - do we still need this?

shrinkray replied:
If we decide we do not need the Framer Motion UI dependency, then we should first delete the component, 

`MobileNav.jsx` then 

`npm uninstall @floating-ui/react framer-motion react-tabs`

 I wanted to wait until we were sure this commit was how we wanted it.

SteamingPilot replied:
Framer, the floating-ui package, and MobileNav.jsx were added by me for a slide-down animation of the navbar on mobile. But Greg didn't end up using the MobileNav, so we can remove the packages and the file.

ShugKnight24 replied:
Removing everything is fine - I was calling out `@floating-ui/react` specifically here because it is not used in the current implementation

Yep, good call on removing `react-tabs`

ShugKnight24 replied:
If we do end up removing these dependencies, please run `npm i` after so we get the updated `package-lock.json`

ShugKnight24 commented on src/App.jsx:
```diff
@@ -7,6 +7,23 @@ import NotFound from '@/pages/NotFound'
     <Router>
+      {/* Skip Link Should be the first element */}
+      <a
+        className={`
+                    absolute 
+                    left-[6px] 
+                    top-[-60px]
+                    z-[100]
+                    bg-white
+                    px-2
+                    py-1
+                    text-black
+                    focus:top-[6px]
+                  `}
```
NIT: Formatting looks off - EOL spaces on 13 & 14 due to single line usage
not important

shrinkray replied:
Am pushing a fix to improve readability and performance.

ShugKnight24 commented on src/pages/Home.jsx:
```diff
@@ -17,14 +19,18 @@ function Home() {
-        speakersData={[]}
-        tracks={[]}
+        speakersData={Speakers2025}
+        tracks={['AI/ML', 'Mobile', 'Fullstack', 'Miscellaneous']}
         defaultExpanded
       />
       <SponsorsSection />
       <OrganizersSection />
       <FacilitatorsSection />
-      <SpeakersSection year={2025} speakersData={[]} defaultExpanded />
+      <SpeakersSection
+        year={2025}
+        speakersData={Speakers2025}
+        defaultExpanded
+      />
```
Previously @JustinBeaudry wanted me to remove the mock data [here](https://github.com/GDG-Detroit/devfest-website/pull/5) - In the resolved conversation. This prevents the call the speakers from appearing in these sections. I'll let him make that call

shrinkray replied:
I missed this conversation. I'll let y'all address it.

SteamingPilot replied:
I added the mock data because the active state link was causing some issues due to the very short height of the speaker and session section. We can remove it since the current handleScroll no longer uses the IoU algorithm. I doubt the new navbar-distance algorithm will have any issues with it.

ShugKnight24 replied:
Just tested, the active state is no longer an issue with the new system if the data isn't there

ShugKnight24 commented on src/layouts/SessionsSection.jsx:
```diff
@@ -82,6 +82,9 @@ const SessionsSection = ({
           className="h-12 md:h-14 lg:h-16"
         />
         <button
+          aria-label={
+            isExpanded ? `Collapse ${year} Sessions` : `Expand ${year} Sessions`
+          }
```
thank you

ShugKnight24 commented on src/layouts/SpeakersSection.jsx:
```diff
@@ -48,6 +48,11 @@ const SpeakersSection = ({
             className="h-8 md:h-10 lg:h-12"
           />
           <button
+            aria-label={
+              isExpanded
+                ? `Collapse ${year} Speakers`
+                : `Expand ${year} Speakers`
+            }
```
👍

shrinkray commented:
@ShugKnight24 @SteamingPilot asked me to push some work I added to help complete the work. Apologies, I went overboard but this may have been needed to get us closer for launch soon. 

1. Fix hardcode navbar height

Original

```
// Original code had hardcoded 96px navbar height
const hardcoded_navbar_height = 96
const navbar_height = hardcoded_navbar_height
```

New

```
// Added dynamic navbar height calculation
const getNavbarHeight = () => {
  const navbar = document.querySelector('nav')
  return navbar ? navbar.offsetHeight : 96
}

// Used throughout navigation and scroll detection
const navbar_height = getNavbarHeight()
```

2. Fixed Active Link Mismatch
Original
- Click nav would go to section
- Scroll detection overrides showing wrong active link
- visual mismatch

New
- immediate feedback
- prevent conflicts


```
// Added manual navigation state to prevent scroll detection interference
const [isManualNavigation, setIsManualNavigation] = useState(false)

const handleNavigation = (event, sectionId) => {
  // Set active link immediately
  setActiveLink(sectionId)
  
  // Prevent scroll detection from overriding
  setIsManualNavigation(true)
  
  // Re-enable after scroll animation completes
  setTimeout(() => {
    setIsManualNavigation(false)
  }, 1000)
}

const handleScroll = () => {
  // Skip scroll detection during manual navigation
  if (isManualNavigation) return
  // ... rest of scroll logic
}
```

3. Mobile nav
- used setTimeout hacks
- chrome specific issues with touch events

New
- reliable
- chrome optimized
- better performance
- accessibility - use touch target size 44px minimum

4. Multi-click issues
- rapid clicking caused unwanted scrolling
- many setTimeout callbacks

New
- Debounced Nav (only one at a time)
- Avoid unnecessary scolling
- Prevent scroll animation conflicts

```
// Added navigation state protection
const [isNavigating, setIsNavigating] = useState(false)

const handleNavigation = (event, sectionId) => {
  // Prevent multiple rapid clicks
  if (isNavigating) {
    return
  }

  setIsNavigating(true)
  
  // Smart distance check - if already close, just update active link
  const distanceFromTarget = Math.abs(targetRect.top - navbar_height)
  if (distanceFromTarget < 100 && !isNavVisible) {
    setActiveLink(sectionId)
    return
  }

  // Re-enable navigation after scroll completes
  setTimeout(() => {
    setIsNavigating(false)
  }, 500) // Shorter timeout for desktop
}
```
5. Separate concerns
Original
- single handler for mobile and desktop
- mobile-specific delays
- complex conditional logic

New
- optimize performance
- optimized path
- easier to maintain

6. Mobile nav UI
- basic styling
- no visual feedback for active state
- only open icon

New
- clean visual feedback
- touch-friendly (larger targets
- follow existing mobile nav patterns

7. Scroll detection
- section switching was offset

New
- prioritize visible sections
- 50px buffer prevents premature switching

8. Production ready cleanup
- maintainable
- no debugging code

shrinkray committed:
feat: add handleHomeNavigation

- click logo to activate
- fits anticipated UX pattern
- mobile menu closes
- return home
- smooth animation
- state management

ShugKnight24 commented on src/data/2025/speakers.js:
```diff
@@ -0,0 +1,136 @@
+// TODO: Replace the following dummy data with real spearkes data.
```
This duplicates src/data/speakers.js

SteamingPilot replied:
Oh my bad. I thought we were using a folder for each year, and left the root files as templates.

ShugKnight24 replied:
No worries - I was following the old site's structure, where prior year data was archived in folders and current was in `/data`. We don't have to keep it that way

ShugKnight24 commented on src/components/Navbar.jsx:
```diff
@@ -72,13 +112,70 @@ function Navbar() {
     if (target) {
-      target.scrollIntoView({ behavior: 'smooth' })
+      // Check if we're already at this section (within a reasonable distance)
+      const navbar_height = getNavbarHeight()
```
NIT: Pythonic variable name. You been working with 🐍 lately? ;)
/s

shrinkray replied:
Am pushing a fix, great catch. Not sure where that came from.

ShugKnight24 reviewed (commented):
Great work @shrinkray & @SteamingPilot. This updated Nav looks much cleaner IMO

Only real questions are about the unused dependency and surrounding the mock data on the home page. I'll defer to @JustinBeaudry

ShugKnight24 commented:
> @ShugKnight24 @SteamingPilot asked me to push some work I added to help complete the work. Apologies, I went overboard but this may have been needed to get us closer for launch soon.

YOU SHOULD BE SORRY @shrinkray!!! Lol, why are you apologizing? This was really thorough and the end result improves the site

shrinkray committed:
refactor: fix typo on const name

shrinkray committed:
refactor: improve readability

- improve formatting
- no lint errors
- readability, single line easier to read
- no template literal overhead

ShugKnight24 commented on src/components/Navbar.jsx:
```diff
@@ -72,13 +126,70 @@ function Navbar() {
+        const targetRect = target.getBoundingClientRect()
+        const y = targetRect.top + window.pageYOffset - navbar_height
+
+        window.scrollTo({ top: y, behavior: 'smooth' })
```
Took a look at this again - I think there is a little room for refactoring to DRY this out if you're feeling up to it

Something like creating a function to calculate the scroll position and a function to handle the navigation state reset and account for the delay. Pass the result of the scroll position calculation to `window.scrollTo` & then reset the navigation

shrinkray replied:
I'll wait till tomorrow ... am into something else now. Too much context switching.

shrinkray replied:
Heck yeah, I could't drop it.

ShugKnight24 replied:
Nice refactor, looks clean

shrinkray committed:
refactor: resolve duplication

- DRY priniciples
- each function clear purpose
- maintenance ease
- readability
- reusability, testing,
- reduce fail risk

shrinkray committed:
refactor: improve appeal

- add effect on click
- reverse on deactivate
- smooth 300ms duration with ease-in-out timing
- no layout shift

shrinkray committed:
refactor: apply aria-current

- adds indication of current page/section for screen readers

shrinkray committed:
refactor: add aria-expanded and labels

- missing state indication
- add aria-labelledby

shrinkray committed:
refactor: add role attr

- add role = "menubar" and role="menuitem"

shrinkray committed:
refactor: add screen reader announcements

- add aria-live region for dynamic content announcements

shrinkray commented:
This last batch addresses Aria fixes that are nav based.

JustinBeaudry committed:
Merge branch '2025' into mobile-navbar-fix

shrinkray committed:
chore: remove dependencies

- remove MobileNav.jsx
- remove framer motion dependencies
- no linting errors
- build successful

JustinBeaudry reviewed (approved):
LGTM!
